swift/apple/FirezoneKit/Sources/FirezoneKit/Helpers/LogCompressor.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  LogCompressor.swift
+//
+//
+//  Created by Jamil Bou Kheir on 3/28/24.
+//
+
+import AppleArchive
+import Foundation
+import System
+
+struct LogCompressor {
+  private let logger: AppLogger
+  private let destinationURL: URL?
+  let fileName: String
+
+  init(logger: AppLogger, destinationURL: URL? = nil) {
+    let dateFormatter = ISO8601DateFormatter()
+    dateFormatter.formatOptions = [.withFullDate, .withTime, .withTimeZone]
+    let timeStampString = dateFormatter.string(from: Date())
+    self.fileName = "firezone_logs_\(timeStampString).aar"
+    self.logger = logger
+    self.destinationURL = destinationURL
+  }
+
+  public func compressFolder(destinationURL: URL? = nil) async throws {
+    try await compressFolderReturningURL(destinationURL: destinationURL)
+  }
+
+  @discardableResult
+  public func compressFolderReturningURL(destinationURL: URL? = nil) async throws -> URL? {
+    guard let logFilesFolderURL = SharedAccess.logFolderURL,
+      let logFilesFolderPath = FilePath(logFilesFolderURL)
+    else {
+      throw SettingsViewError.logFolderIsUnavailable
+    }
+
+    let fileManager = FileManager.default
+    let fileURL =
+      destinationURL
+      ?? fileManager.temporaryDirectory.appendingPathComponent(fileName)
+
+    // Remove logfile if it happens to exist
+    if fileManager.fileExists(atPath: fileURL.path) {
+      try fileManager.removeItem(at: fileURL)
+    }
```

Comment:
If we ignore errors anyway maybe this is sufficient?
```suggestion
    try fileManager.removeItem(at: fileURL)
```

It would close a TOCTOU gap

Reply:
This will always fail though, because this file will have a very low likelihood of being around given the timestamp.

We could just remove this, and overwrite in the rare case.

Reply:
The `?` version does the trick.

```
try? fileManager.removeItem(at: fileURL)
```

Reply:
I assumed `try fileManager.removeItem` means we ignore the error if we can't remove it. If it bails out, then yeah keep it as-is